x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/RuleQueryBuilder.java

```diff
@@ -241,31 +283,38 @@ private List<?> truncateList(List<?> input) {
     protected boolean doEquals(RuleQueryBuilder other) {
         if (this == other) return true;
         if (other == null || getClass() != other.getClass()) return false;
-        return Objects.equals(rulesetId, other.rulesetId)
+        return Objects.equals(rulesetIds, other.rulesetIds)
             && Objects.equals(matchCriteria, other.matchCriteria)
             && Objects.equals(organicQuery, other.organicQuery)
-            && Objects.equals(pinnedIds, other.pinnedIds)
-            && Objects.equals(pinnedDocs, other.pinnedDocs)
             && Objects.equals(pinnedIdsSupplier, other.pinnedIdsSupplier)
             && Objects.equals(pinnedDocsSupplier, other.pinnedDocsSupplier);
     }
 
     @Override
     protected int doHashCode() {
-        return Objects.hash(rulesetId, matchCriteria, organicQuery, pinnedIds, pinnedDocs, pinnedIdsSupplier, pinnedDocsSupplier);
+        return Objects.hash(rulesetIds, matchCriteria, organicQuery, pinnedIdsSupplier, pinnedDocsSupplier);
     }
 
     private static final ConstructingObjectParser<RuleQueryBuilder, Void> PARSER = new ConstructingObjectParser<>(NAME, a -> {
         QueryBuilder organicQuery = (QueryBuilder) a[0];
         @SuppressWarnings("unchecked")
         Map<String, Object> matchCriteria = (Map<String, Object>) a[1];
         String rulesetId = (String) a[2];
-        return new RuleQueryBuilder(organicQuery, matchCriteria, rulesetId);
+        @SuppressWarnings("unchecked")
+        List<String> rulesetIds = (List<String>) a[3];
+        if (rulesetId == null && rulesetIds == null) {
+            throw new IllegalArgumentException("no ruleset specified");
+        }
+        if (rulesetIds == null) {
+            rulesetIds = List.of(rulesetId);
+        }
```

Comment:
I think the parser should throw if both `ruleset_id` and `ruleset_ids` are specified.
WDYT?

Also should we use the deprecation logger when using `ruleset_id`?

Reply:
Updated!

Reply:
> I think the parser should throw if both ruleset_id and ruleset_ids are specified.

Not sure if this is handled, but it looks like we don't have a test for this?

Reply:
Good catch!